config/auth_permutations_test.go

```diff
@@ -473,3 +497,62 @@ func TestConfig_AuthTypeFromEnv(t *testing.T) {
 		AssertHost: "https://x",
 	}.apply(t)
 }
+
+func TestConfig_MetadataServerInvalidAuthServerUrl(t *testing.T) {
+	configFixture{
+		AuthType:           "metadata-service",
+		MetadataServiceURL: "not-valid-hostname",
+		Host:               "x",
+		AssertError:        `default auth: metadata-service: invalid auth server URL: not-valid-hostname. Config: host=https://x, metadata_service_url=***`,
+	}.apply(t)
+}
+
+func TestConfig_MetadataServerInvalidAuthServerUrl2(t *testing.T) {
+	configFixture{
+		AuthType:           "metadata-service",
+		MetadataServiceURL: "https://google.com",
+		Host:               "x",
+		AssertError:        `default auth: metadata-service: invalid auth server URL: https://google.com. Config: host=https://x, metadata_service_url=***`,
+	}.apply(t)
+}
+
+func TestConfig_MetadataServerConfigure(t *testing.T) {
+	configFixture{
+		AuthType:           "metadata-service",
+		MetadataServiceURL: "http://localhost/token",
+		Host:               "x",
+		AssertHost:         "https://x",
+		AssertAuth:         "metadata-service",
+		AssertHeaders: map[string][]string{
```

Comment:
```suggestion
		AssertHeaders: map[string]string{
```
this way it's easier to port across languages

Reply:
This is the underlying type of `http.Header`. Other languages also follow this same pattern of headers being a Map[String, Iterable[String]] (https://javaee.github.io/javaee-spec/javadocs/javax/servlet/http/HttpServletRequest.html#getHeaders-java.lang.String-, https://pytest-httpserver.readthedocs.io/en/latest/api.html#pytest_httpserver.RequestHandler.respond_with_data), so I was mainly following this standard.